### ISSUE TITLE
UDP client does not work when server is deployed to a remote location

### ISSUE THREAD

**Issue by RobinTo** [tom-weiland/tcp-udp-networking#2]
# Summary

When you deploy the game server to a remote ip, the UDP client does not connect successfully.

This is due to this line
https://github.com/tom-weiland/tcp-udp-networking/blob/master/GameClient/Assets/Scripts/Client.cs#L207
always instantiating the ip endpoint to `127.0.0.1`, since it fires before the assigning of the ip address on the instance.

I just spent quite a few hours going down a socket debugging rabbithole, when the error was in fact super simple, mainly due to the lack of helpful error messages(and lack of knowledge I suppose), so I'm hoping it can be helpful to you or others.

## Reproduction

- Start the server on any other machine than localhost
- Attempt to connect with client
- UDP client instantly disconnects, and in turn, the client disconnects.

## Proposed solution

Personally I moved the line in question into the connect function, and it appears to work as expected. It just needs to be called after the client instance ip has been set.

## Note

Thank you for the repo and videos, it has been really helpful in getting a basic setup working quickly.

**Comment by kascodingNL** [comment]
You said it was always instantiating the ip endpoint to 127.0.0.1,
that is because instance.ip is "127.0.0.1".
Could you try changing that to the IP of the server?

**Comment by RobinTo** [comment]
> You said it was always instantiating the ip endpoint to 127.0.0.1,
> that is because instance.ip is "127.0.0.1".
> Could you try changing that to the IP of the server?

Yes, that is of course the solution, as I mention.

The issue is redundant though, as the issue does not exist in this version. I implemented a field for IP address when following the tutorials, and I thought it was part of this repo. When looking into it, I notice that it is not. The UDP client did not utilize that value, since it was statically set in the mentioned code, so I just had to override it with the ip field on connect.

So this is a non issue here, as the IP input field isn't part of the repo 🤦‍♂  👍
